### ISSUE TITLE
Don't be sad, there is an Atom's continuous editor - Zed

### ISSUE THREAD

**Issue by zhaolinlau** [atom/atom#25511]
Zed is a fast, collaborative code editor created by the same team that brought you GitHub's Atom editor. Also, Zed is currently in full-time development, links to follow up of any news about Zed:
https://news.ycombinator.com/item?id=31669615
https://zed.dev/
https://twitter.com/zeddotdev
https://twitter.com/nathansobo

**Comment by a-chacon** [comment]
Do you know when we will be able to try zed??

**Comment by zhaolinlau** [comment]
> Do you know when we will be able to try zed?? 

Alpha version will be released by this week privately, you may subscribe to get involved early through [zed.dev](https://zed.dev/)

**Comment by iamunadike** [comment]
We just need ATOM this is not time to try to sell a different editior People have not vigorously used and tested

**Comment by SavSanta** [comment]
Bro relax. There it is not done til December. You got time. Also, their atom-community  version of atom that atom enthusiast will continue to work on so you should be good

**Comment by a-chacon** [comment]
You are right! But is not bad to have an alternative. And Zed looks like a good. I hope the community version of Atom continue too.

**Comment by UziTech** [comment]
The biggest problem with any new editor is it takes time for third party extensions to get created. I would guess zed will be as good as atom about a year after it is released.

**Comment by UziTech** [comment]
Hopefully zed will be completely customizable like atom (with an init.js and custom style sheet). Otherwise it will be no better that vscode in my mind. I don't need an editor that loads a couple milliseconds faster, I need one that I can fully customize, one that works the way I want it to.

**Comment by jonian** [comment]
Zed will not be open source, so not an alternative to Atom.

**Comment by fanninpm** [comment]
For those who are looking for an open-source alternative to Zed, [Lapce](https://github.com/lapce/lapce) may be worth taking a look at.

**Comment by raer0** [comment]
> Zed is a fast, collaborative code editor created by the same team that brought you GitHub's Atom editor. Also, Zed is currently in full-time development, links to follow up of any news about Zed: https://news.ycombinator.com/item?id=31669615 https://zed.dev/ https://twitter.com/zeddotdev https://twitter.com/nathansobo



> Zed will not be open source, so not an alternative to Atom.

@zhaolinlau Please elaborate as to how you believe Zed to be a valid alternative to Atom, when it is not FOSS? Please do not attempt to take advantage of a dying open source project to promote a proprietary project.

**Comment by DeeDeeG** [comment]
@jonian @raer0 The code for Zed has not been shared, _yet_. It is still pre-alpha (or private alpha some time this month). Nathan Sobo [said](https://youtu.be/wXT73bBr83s?t=42m18s) in a talk that in the long-term they want the core editor to be "**free and probably open-source**." (Sounds like/I guess implies the _collaboration feature_ would be monetized and _potentially_ closed-source, while clarifying that the collaboration feature would probably be an _optional_ part of using the editor. Even if it is a very important reason to the creators motivating them to build this app.)

Zed is not here yet. Best not to speculate, or assert things that aren't set in stone yet. If you want an update on Zed, this talk is pretty much the only major update recently: https://youtu.be/wXT73bBr83s

**Comment by jonian** [comment]
@DeeDeeG It looks from [this](https://news.ycombinator.com/item?id=31669814) that it will be closed source. Maybe they will open some parts, like the ui framework or whatever. Nothing is clear about that project so there is no need to promote it here. If people want to suggest alternatives there are many open source projects.

- Lite XL https://github.com/lite-xl/lite-xl
- Lapce https://github.com/lapce/lapce
- Builder https://gitlab.gnome.org/GNOME/gnome-builder
- Text Adept https://github.com/orbitalquark/textadept
- Geany https://github.com/geany/geany
- Gedit https://gitlab.gnome.org/GNOME/gedit
- Code https://github.com/elementary/code
- Kate https://kate-editor.org
- etc...

**Comment by mikajed** [comment]
> Zed will not be open source, so not an alternative to Atom.

If you watch the video of the developer, he said the "single player mode" will be open source, the co-op coding mode will not.

**Comment by chfritz** [comment]
So we just went through the experience of losing our favorite editor after getting used to for many years and it was backed by freaking Github and you think we might be in the mood to bet our new muscle memory and customization efforts on an editor that is not even release yet *nor* is open-source *nor* is backed by a big company? I don't think mourning Atom users are your target audience, @zhaolinlau .

**Comment by Ruzie** [comment]
> Zed is a fast, collaborative code editor created by the same team that brought you GitHub's Atom editor. Also, Zed is currently in full-time development, links to follow up of any news about Zed: https://news.ycombinator.com/item?id=31669615 https://zed.dev/ https://twitter.com/zeddotdev https://twitter.com/nathansobo

YO! where is the source code or is it yet another proprietary software?

**Comment by autumnblazey** [comment]
Hm. Will consider if it becomes _fully and completely open source_, otherwise I do not have any incentive to switch over. I have buried myself in the vscode ecosystem face first, as much as I loathe that I have done that and bought myself into something microsoft. You will have to prove yourself, and honestly, my impression of this editor is not that great, seeing as I cannot see the code and follow development, and have _already_ seen multiple adverts everywhere around atom for it.

(If wondering, I did switch to vscode one or two years ago, but I still love atom, it being the only other text editor I have used at all for years and years, and was always looking for every single opportunity to switch back if even partially and be able to do stuff with it again. Was just waiting for apple silicon support, honestly)

**Comment by elidrissidev** [comment]
That image doesn't look promising
![Screen Shot 2022-06-13 at 5 52 46 PM](https://user-images.githubusercontent.com/67818913/173405007-b732e179-750a-4ffc-82d9-231ae8d8d7f5.png)

**Comment by SoftOrangeTech** [comment]
**It says on the website** that it is on development WTF is he talking about?!?!?!?!?!

I'm not getting involved in their development team.

**Comment by SoftOrangeTech** [comment]
> More Detailed

The website clearly says that it's on the development team. We might even sue github for this. GitHub does not make the great decision. We are months left from losing Atom. I will make a special thing that can make them reinstate Atom. @github is not good.

**Comment by SoftOrangeTech** [comment]
I mentioned GitHub/

**Comment by Digitalone1** [comment]
LOL, next generation? It's not even open source.

**Comment by hermidalc** [comment]
I have been very pro Atom since I started using it in 2017 until last week. Loved every minute of it, best IDE I've ever used. But have to get my job done and not waiting until last minute until more Atom breaks.

I just moved to [VSCodium](https://vscodium.com/) because I need to get my job done TODAY, can't wait for pre-alpha or beta madness right now. VSCode has a ton of settings and takes some time to get adjusted to, but in only a few days I feel pretty comfortable. Honestly it's just a tool for me to get my job done effectively (scientific software dev and comp analysis in python and R) and I don't care that much am not a cultist, and guess what it actually works well and has features I've always wanted that Atom never had.

A feature VSCode has that Atom didn't and it's a huge plus for me is that each your project folders can be associated with their respective mamba/conda environments - so each with their own python and libraries specific to that project. It automatically switches depending on the file in focus in the editor. This is how Atom should've been done.

I'm still trying to figure out how to do the the analogous with R in VSCode, i.e. automatically attaching the project folder conda env R based on the file in focus in the editor, but can't see to figure that out how right now, though at least the framework is there because on the python side it just works.

And because of this in Linux I just launch VSCodium from the GNOME desktop, not like I had to do with Atom where I opened a terminal, activated a conda env, launched Atom and then was stuck with that conda env and not automatic env switching based on the file in focus.

**Comment by jonian** [comment]
@hermidalc People that like Atom more than VS Code don't care if VS Code is better or faster. VS Code is not as hackable as Atom and that is their biggest difference. It is better to move to Neovim, Vim, Emacs or any other open source editor developed by a community and not a corporation that might decide to sunset their editor at any time. 

New github CEO said, 4 years ago, that he understands that the code editor is a personal choice, difficult to switch, etc and that they will keep developing atom. 4 years from now they might decide to sunset VS Code in favor of a new proprietary editor or a cloud editor and you will have to switch editors again.

The worst part of VS Code is that it is infested with telemetry. VS Codium is playing catch up with Microsoft to disable telemetry on every new version. Check their github repo, they still have open issues related to telemetry. 

If you want an editor that looks like VS Code and is compatible with with it's extensions then use [Theia](https://theia-ide.org). It is not a fork of VS Code and it is a truly open source project.

> Theia is truly vendor-neutral and developed by a diverse community. Unlike other “open source” projects, Theia is actually hosted at an Open-Source Foundation, protecting it against single-vendor decisions that are against the interest of the community.

**Comment by iamunadike** [comment]
> @hermidalc People that like Atom more than VS Code don't care if VS Code is better or faster. VS Code is not as hackable as Atom and that is their biggest difference. It is better to move to Neovim, Vim, Emacs or any other open source editor developed by a community and not a corporation that might decide to sunset their editor at any time.
> 
> New github CEO said, 4 years ago, that he understands that the code editor is a personal choice, difficult to switch, etc and that they will keep developing atom. 4 years from now they might decide to sunset VS Code in favor of a new proprietary editor or a cloud editor and you will have to switch editors again.
> 
> The worst part of VS Code is that it is infested with telemetry. VS Codium is playing catch up with Microsoft to disable telemetry on every new version. Check their github repo, they still have open issues related to telemetry.
> 
> If you want an editor that looks like VS Code and is compatible with with it's extensions then use [Theia](https://theia-ide.org). It is not a fork of VS Code and it is a truly open source project.
> 
> > Theia is truly vendor-neutral and developed by a diverse community. Unlike other “open source” projects, Theia is actually hosted at an Open-Source Foundation, protecting it against single-vendor decisions that are against the interest of the community.

The best answer I have got since all these threads, a man that truly understands whats going on. fact is , I feel since atom released 1.60 after a long while of free developers maintaining it, which was late last year , Micro  people got jealouss, let m be honest the reason for this archiving is something more sinister than it seems !!!
Never Trust Bill Gates

**Comment by iamunadike** [comment]
Dont forget when you dont have choices , You are more or less being FORCED to use what is available . 
Coughs ....
or let me CORRECT myself, WHAT IS POPULAR AND SINISTER

**Comment by hermidalc** [comment]
I think for 90% of devs the level of hackability available in VSCode is plenty good enough from what I see so far. There's tons of extensions and yes sure you can do some of the quite down to the bolts changes that Atom allowed, but again for 90% of devs they aren't doing that any way.

I'm only currently moving to VSCodium because I need to continue working, when another thing as good as Atom with as good and mature an ecosystem comes up, I'll change to it it only takes like a week or so to get comfortable with a new IDE. No biggie.

**Comment by iamunadike** [comment]
Another thing as good as atom may NEVER come, It will take a long while and a decent move by DEVELOPERS who really Have the TImE and the MoNey

**Comment by hermidalc** [comment]
> Another thing as good as atom may NEVER come, It will take a long while and a decent move by DEVELOPERS who really Have the TImE and the MoNey

EXACTLY... and the rest of us need to continue our jobs and work. So sorry it's hard to swallow the jagged little pill but sometimes you have to until the next thing comes. I can't wait for years or deal with alpha level IDE I don't have the time or energy.

**Comment by iamunadike** [comment]
So maintaining what is available now is a good way to MainTAIN rather than RENEW, because I am not sure others will be open source, Even SUBLIME does not near atom, If Atom was a bad project yet after many years ... Devs still can not do without
Guess what... That is a PRODUCT that can not di

**Comment by hermidalc** [comment]
The problem is it likely will die. I cannot find the “Is atom dead?” issue anymore (what happened to it?) but I believe @aminya who is founder or co-founder of `atom-community` and others have said there that Atom’s core and extension framework or something need a big refactor/redo/redesign.

There are a number of core issues limiting continued Atom development. And like everything in soft dev world, if it’s not possible to evolve anymore then it dies. If it were easier to continue to work on Atom then `atom-community` would’ve done it. They already did an amazing job with `atom-ide-community` when nuclide went bye bye and I remember immediately switching over to it and it just worked great. So I trust what they’ve said.

**Comment by UziTech** [comment]
Talk of an Atom fork are in way on the [atom-community discord](https://discord.gg/d2cY7AjJtG)

**Comment by hermidalc** [comment]
> Talk of an Atom fork are in way on the [atom-community discord](https://discord.gg/d2cY7AjJtG)

It looks like maybe @aminya and co are bringing the existing atom-community Atom [fork](https://github.com/atom-community/atom) back alive. There've been commits to that fork here and there over time, but I swear ppl wrote before that they were giving up on it because Atom fundamentally needed so heavy work to make it easier to continue to develop.

**Comment by autumnblazey** [comment]
I am in there, and there are a lot of discussions going on!

**Comment by manngo** [comment]
I have contacted the developers, and addons are planned for some time in the future. That’s fine for the future, but it’s not yet a viable alternative to Atom in the present.

**Comment by iamunadike** [comment]
personally I stand by atom
Atom is irreplaceable
and we will make it better this time

**Comment by mikajed** [comment]
> personally I stand by atom Atom is irreplaceable and we will make it better this time
While Atom was/is pretty nice, it actually is replaceable, there are so many editors out there like code - oss or even neovim which I started using.

**Comment by hermidalc** [comment]
> personally I stand by atom Atom is irreplaceable and we will make it better this time

Yeah this is a bit of cult-like evangelism, it's just an editor.

Even at it's prime Atom had many flaws and significant missing features at it's core and within it's ecosystem. I loved and used it for 5 years almost every single day, but now that I'm using VSCode I realize how many important features Atom was missing that make my job so much easier. If Atom can come back and include these missing features e.g. like the ones I've described above, like being able to open Atom from the desktop and with the editor setting the Python and R interpreter and associated environment (e.g. conda) for each project folder from a list it finds on your box, instead of having to open Atom in a shell with a conda environment activated and then closing and doing it all over again with another conda env for each damn project, of having native Wayland support instead of XWayland (like VSCode has had for a long time), of having Snakemake language and formatting support (useful for my very large community of devs), etc, then I will come back.

**Comment by iamunadike** [comment]
as a community we have a fork at github.com/atom-community/atom
you too come come and contribute
not easy but its possible

**Comment by MMarcelo85** [comment]
> I have been very pro Atom since I started using it in 2017 until last week. Loved every minute of it, best IDE I've ever used. But have to get my job done and not waiting until last minute until more Atom breaks.
> 
> I just moved to [VSCodium](https://vscodium.com/) because I need to get my job done TODAY, can't wait for pre-alpha or beta madness right now. VSCode has a ton of settings and takes some time to get adjusted to, but in only a few days I feel pretty comfortable. Honestly it's just a tool for me to get my job done effectively (scientific software dev and comp analysis in python and R) and I don't care that much am not a cultist, and guess what it actually works well and has features I've always wanted that Atom never had.
> 
> A feature VSCode has that Atom didn't and it's a huge plus for me is that each your project folders can be associated with their respective mamba/conda environments - so each with their own python and libraries specific to that project. It automatically switches depending on the file in focus in the editor. This is how Atom should've been done.
> 
> I'm still trying to figure out how to do the the analogous with R in VSCode, i.e. automatically attaching the project folder conda env R based on the file in focus in the editor, but can't see to figure that out how right now, though at least the framework is there because on the python side it just works.
> 
> And because of this in Linux I just launch VSCodium from the GNOME desktop, not like I had to do with Atom where I opened a terminal, activated a conda env, launched Atom and then was stuck with that conda env and not automatic env switching based on the file in focus.

Well you could just have installed ipykernel on your envs and then added your env to it. In this way Atom asks to you which env you want to use:

![Selección_288](https://user-images.githubusercontent.com/57275006/192629399-5fd8966c-5e0b-4ef8-8bdb-e0170fecb5ca.png)

**Comment by hermidalc** [comment]
> > I have been very pro Atom since I started using it in 2017 until last week. Loved every minute of it, best IDE I've ever used. But have to get my job done and not waiting until last minute until more Atom breaks.
> > I just moved to [VSCodium](https://vscodium.com/) because I need to get my job done TODAY, can't wait for pre-alpha or beta madness right now. VSCode has a ton of settings and takes some time to get adjusted to, but in only a few days I feel pretty comfortable. Honestly it's just a tool for me to get my job done effectively (scientific software dev and comp analysis in python and R) and I don't care that much am not a cultist, and guess what it actually works well and has features I've always wanted that Atom never had.
> > A feature VSCode has that Atom didn't and it's a huge plus for me is that each your project folders can be associated with their respective mamba/conda environments - so each with their own python and libraries specific to that project. It automatically switches depending on the file in focus in the editor. This is how Atom should've been done.
> > I'm still trying to figure out how to do the the analogous with R in VSCode, i.e. automatically attaching the project folder conda env R based on the file in focus in the editor, but can't see to figure that out how right now, though at least the framework is there because on the python side it just works.
> > And because of this in Linux I just launch VSCodium from the GNOME desktop, not like I had to do with Atom where I opened a terminal, activated a conda env, launched Atom and then was stuck with that conda env and not automatic env switching based on the file in focus.
> 
> Well you could just have installed ipykernel on your envs and then added your env to it. In this way Atom asks to you which env you want to use:
> 
> ![Selección_288](https://user-images.githubusercontent.com/57275006/192629399-5fd8966c-5e0b-4ef8-8bdb-e0170fecb5ca.png)

That's good to know, thank you didn't know could do that. But since now I've been using VSCode heavily for over 3 months there are many more very useful features for me that it has via extensions marketplace that I check and Atom doesn't have. So this list is only growing for me. And instead of waiting for Atom, or dealing with it as it slowly breaks, I've been getting my work done this entire time and really loving VSCode every minute of it. End of this year Atom will get no more updates and I'm not sure the community trying to resurrect it will be ready at that time.

I'm not against Atom in any way, I loved it for years, but getting my job done is much so more important than some stupid telemetry debate. Have you noticed when I commented above that if everyone were so crazy about telemetry against VSCode, then they should all be destroying their Apple and Android devices because no matter how much you try and lock these down with apps you are still getting tracked in an intrusive way. Crickets... no response to my comment, because it's true this IDE telemetry argument becomes a ridiculous when your mobile devices are tracking your personal stuff so much more!

**Comment by MMarcelo85** [comment]
> > > I have been very pro Atom since I started using it in 2017 until last week. Loved every minute of it, best IDE I've ever used. But have to get my job done and not waiting until last minute until more Atom breaks.
> > > I just moved to [VSCodium](https://vscodium.com/) because I need to get my job done TODAY, can't wait for pre-alpha or beta madness right now. VSCode has a ton of settings and takes some time to get adjusted to, but in only a few days I feel pretty comfortable. Honestly it's just a tool for me to get my job done effectively (scientific software dev and comp analysis in python and R) and I don't care that much am not a cultist, and guess what it actually works well and has features I've always wanted that Atom never had.
> > > A feature VSCode has that Atom didn't and it's a huge plus for me is that each your project folders can be associated with their respective mamba/conda environments - so each with their own python and libraries specific to that project. It automatically switches depending on the file in focus in the editor. This is how Atom should've been done.
> > > I'm still trying to figure out how to do the the analogous with R in VSCode, i.e. automatically attaching the project folder conda env R based on the file in focus in the editor, but can't see to figure that out how right now, though at least the framework is there because on the python side it just works.
> > > And because of this in Linux I just launch VSCodium from the GNOME desktop, not like I had to do with Atom where I opened a terminal, activated a conda env, launched Atom and then was stuck with that conda env and not automatic env switching based on the file in focus.
> > 
> > 
> > Well you could just have installed ipykernel on your envs and then added your env to it. In this way Atom asks to you which env you want to use:
> > ![Selección_288](https://user-images.githubusercontent.com/57275006/192629399-5fd8966c-5e0b-4ef8-8bdb-e0170fecb5ca.png)
> 
> That's good to know, thank you didn't know could do that. But since now I've been using VSCode heavily for over 3 months there are many more very useful features for me that it has via extensions marketplace that I check and Atom doesn't have. So this list is only growing for me. And instead of waiting for Atom, or dealing with it as it slowly breaks, I've been getting my work done this entire time and really loving VSCode every minute of it. End of this year Atom will get no more updates and I'm not sure the community trying to resurrect it will be ready at that time.
> 
> I'm not against Atom in any way, I loved it for years, but getting my job done is much so more important than some stupid telemetry debate. Have you noticed when I commented above that if everyone were so crazy about telemetry against VSCode, then they should all be destroying their Apple and Android devices because no matter how much you try and lock these down with apps you are still getting tracked in an intrusive way. Crickets... no response to my comment, because it's true this IDE telemetry argument becomes a ridiculous when your mobile devices are tracking your personal stuff so much more!

I agree with you 100%. I also installed VScode it has terrific add-ons. Still, I think I'm riding with Atom until the end, Jaja. vscode is great, although I  miss Hydrogen when I'm using VScode, and I have not found anything alike. Also, vscode needs some configuration that I need to do and some of my macros are not responding anymore, so I need to get to know it better, I didn't find the time nor the will... I guess it's a matter of time. About telemetry, everyone is using google, social media and, everyone has a phone, so I don't get it too. Also, I think you can shut it down on preferences.

**Comment by manngo** [comment]
As regards VSCode:

>About telemetry, everyone is using google, social media and, everyone has a phone, so I don't get it too. Also, I think you can shut it down on preferences.

Apparently, you can’t turn of telemetry. And personally I don’t use Google, social media, or anything which increases my chances of being tracked.

If I go down the BS path, it will be  VSCodium because it’s not Microsoft. I strongly suspect it’s Microsoft, the overlords of GitHub, who is behind giving Atom the shove

**Comment by aminya** [comment]
I think it is essential for me to clarify the status of the Atom community fork.

We still get some occasional contributions. We set the bar very low for the contributions. As long as the unit tests pass in the Azure CI, we will accept the pull requests. 

However, to be honest, the project lacks dedicated and experienced developers that are willing to contribute to the project regularly. As you know, most of the previous developers have moved on to develop **Zed**, and the remaining are among the occasional contributors. Given the current situation, I don't think we will see any major changes in Atom.

**Comment by hermidalc** [comment]
> As regards VSCode:
> 
> > About telemetry, everyone is using google, social media and, everyone has a phone, so I don't get it too. Also, I think you can shut it down on preferences.
> 
> Apparently, you can’t turn of telemetry. And personally I don’t use Google, social media, or anything which increases my chances of being tracked.

I guess you have a flip phone too, either way n=1, you are a rarity among the users here complaining about VSCode telemetry. Atom is never really coming back because as @aminya said they don't have serious developers taking the reigns and the ecosystem will very quickly dry up after 2022.

> If I go down the BS path, it will be VSCodium because it’s not Microsoft. I strongly suspect it’s Microsoft, the overlords of GitHub, who is behind giving Atom the shove.

Of course they did we all knew it was coming once they bought GitHub, it was just a matter of time. I never believed what they said about continuing Atom dev and anyone who did is naive. I stayed with Atom until this summer but things started to break and weren't getting fixed, and knew at some point at had to make the switch to another IDE and VSCode/Codium are really the only thing in town right now truly worth using.

If Zed ever makes it to Atom level I would love to try it out.

**Comment by aminya** [comment]
Those who want to see the progress for Zed can follow these links:

https://github.com/zed-industries/feedback
https://zed.dev/releases
